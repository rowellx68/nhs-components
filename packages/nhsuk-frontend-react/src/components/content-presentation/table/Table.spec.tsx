import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  TwoColumn as TwoColumnStory,
  ThreeColumn as ThreeColumnStory,
} from './Table.stories';

const TwoColumn = composeStory(TwoColumnStory, meta);
const ThreeColumn = composeStory(ThreeColumnStory, meta);

it('should render a two column table', () => {
  const { container } = render(<TwoColumn />);

  expect(container.querySelector('caption')).toHaveTextContent(
    'Skin symptoms and possible causes',
  );
  expect(container).toMatchSnapshot();
});

it('should render a three column table', () => {
  const { container } = render(<ThreeColumn />);

  expect(container.querySelector('caption')).toHaveTextContent(
    'Ibuprofen tablet dosages for children',
  );
  expect(
    container.querySelector('table.nhsuk-table-responsive'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
