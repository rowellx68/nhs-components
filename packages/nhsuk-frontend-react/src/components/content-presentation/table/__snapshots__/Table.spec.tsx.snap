// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a three column table 1`] = `
<div>
  <table
    class="nhsuk-table-responsive"
  >
    <caption
      class="nhsuk-table__caption"
    >
      Ibuprofen tablet dosages for children
    </caption>
    <thead
      class="nhsuk-table__head"
      role="rowgroup"
    >
      <tr
        class=""
        role="row"
      >
        <th
          data-responsive-heading="Age"
          role="columnheader"
          scope="col"
        >
          Age
        </th>
        <th
          data-responsive-heading="How much?"
          role="columnheader"
          scope="col"
        >
          How much
        </th>
        <th
          data-responsive-heading="How often?"
          role="columnheader"
          scope="col"
        >
          How often
        </th>
      </tr>
    </thead>
    <tbody
      class="nhsuk-table__body"
    >
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            Age
             
          </span>
          7 to 9 years
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How much?
             
          </span>
          200mg
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How often?
             
          </span>
          Max 3 times in 24 hours
        </td>
      </tr>
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            Age
             
          </span>
          10 to 11 years
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How much?
             
          </span>
          200mg to 300mg
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How often?
             
          </span>
          Max 3 times in 24 hours
        </td>
      </tr>
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            Age
             
          </span>
          12 to 17 years
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How much?
             
          </span>
          200mg to 400mg
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          <span
            aria-hidden="true"
            class="nhsuk-table-responsive__heading"
          >
            How often?
             
          </span>
          Max 3 times in 24 hours
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render a two column table 1`] = `
<div>
  <table
    class="nhsuk-table"
  >
    <caption
      class="nhsuk-table__caption"
    >
      Skin symptoms and possible causes
    </caption>
    <thead
      class="nhsuk-table__head"
      role="rowgroup"
    >
      <tr
        class=""
        role="row"
      >
        <th
          role="columnheader"
          scope="col"
        >
          Skin symptoms
        </th>
        <th
          role="columnheader"
          scope="col"
        >
          Possible cause
        </th>
      </tr>
    </thead>
    <tbody
      class="nhsuk-table__body"
    >
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          Blisters on lips or around the mouth
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          cold sores
        </td>
      </tr>
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          Itchy, dry, cracked, sore
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          eczema
        </td>
      </tr>
      <tr
        class="nhsuk-table__row"
        role="row"
      >
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          Red, scaly patches
        </td>
        <td
          class="nhsuk-table__cell"
          role="cell"
        >
          psoriasis
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
